bindings/rust/generate/src/main.rs

```diff
@@ -0,0 +1,142 @@
+use std::path::{Path, PathBuf};
+use std::collections::HashSet;
+use std::{fs, io};
+use std::fs::File;
+use std::io::{BufRead, BufReader, BufWriter, Write};
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use regex::Regex;
+
+/*
+// See PR: https://github.com/rust-lang/rust-bindgen/pull/2228
+
+#[derive(Debug)]
+struct SymbolCallback {
+    prefix: String,
+}
+
+impl SymbolCallback {
+    fn new() -> Self {
+        SymbolCallback {
+            prefix: format!("{}_", prefix_string()),
+        }
+    }
+}
+
+impl bindgen::callbacks::ParseCallbacks for SymbolCallback {
+    fn generated_name_override(&self, function_name: &str) -> Option<String> {
+        let mut result = function_name.to_string();
+        if result.starts_with(&self.prefix) {
+            result = result.replace(&self.prefix, "");
+            Some(result)
+        } else {
+            None
+        }
+    }
+}
+*/
+
+fn modify_bindings(bindings_path: &PathBuf, prefix: &str) -> io::Result<()>{
+    // Needed until this issue is resolved: https://github.com/rust-lang/rust-bindgen/issues/1375
+
+    let prefix_func_detector = Regex::new(&format!("(^\\s+)pub\\s+fn\\s+{}_(\\w*)(.*)", prefix)).unwrap();
```

Comment:
I think this regex is complicated enough to warrant a comment briefly explaining it, especially since it's also doing string formatting.

Reply:
Good point. Will do.